include/zephyr/data/json.h

```diff
@@ -109,7 +109,8 @@ struct json_obj_descr {
 	union {
 		struct {
 			const struct json_obj_descr *sub_descr;
-			size_t sub_descr_len;
+			uint16_t sub_descr_len;
+			uint16_t sub_struct_size;
```

Comment:
what's the reason for the type change and why did you choose `uint16_t`?

Reply:
Reverted the type for `sub_descr_len` to `size_t`.

I do agree, I moved the field `sub_struct_size` up and renamed it to `struct_size`. It just required a little more work to have all descriptors defined correctly: I needed to explicitly define the `struct_size` member to an arbitrary value (e.g. 0).